python/equistore-core/equistore/core/labels.py

```diff
@@ -456,6 +456,105 @@ def position(self, entry: Union[LabelsEntry, Sequence[int]]) -> Optional[int]:
         else:
             return None
 
+    def union(self, other: "Labels") -> "Labels":
+        """Take the union of these :py:class:`Labels` with ``other``."""
```

Comment:
I would add an see also here to link to the `union_and_mapping` if one also wants to have the mapping. Same for the intersection.